server/resp/src/main/java/org/infinispan/server/resp/json/JsonDocGetFunction.java

```diff
@@ -0,0 +1,160 @@
+package org.infinispan.server.resp.json;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.functional.EntryView;
+import org.infinispan.functional.EntryView.ReadWriteEntryView;
+import org.infinispan.server.resp.ExternalizerIds;
+import org.infinispan.util.function.SerializableFunction;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter.Indenter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jayway.jsonpath.JsonPath;
+
+public class JsonDocGetFunction
+      implements SerializableFunction<EntryView.ReadWriteEntryView<byte[], JsonDocBucket>, String> {
+   public static final AdvancedExternalizer<JsonDocGetFunction> EXTERNALIZER = new JsonDocGetFunction.Externalizer();
+   static final byte[] EMPTY_BYTES = new byte[0];
+   public static final String ERR_PATHS_CAN_T_BE_NULL = "paths can't be null";
+
+   byte[] space;
+   byte[] newline;
+   byte[] indent;
+   List<byte[]> paths;
+
+   public JsonDocGetFunction(List<byte[]> paths, byte[] space, byte[] newline, byte[] indent) {
+      requireNonNull(paths, ERR_PATHS_CAN_T_BE_NULL);
+      this.paths = paths;
+      this.space = (space != null) ? space : EMPTY_BYTES;
+      this.newline = (newline != null) ? newline : EMPTY_BYTES;
+      this.indent = (indent != null) ? indent : EMPTY_BYTES;
+   }
+
+   @Override
+   public String apply(ReadWriteEntryView<byte[], JsonDocBucket> entryView) {
```

Comment:
One thing which is debatable is whether we want to return the "pretty" version or leave to the caller to do the work. Having the caller do the work could shave a few bytes for remote calls, but I guess it could be a future PR.

Reply:
yeah, iirc I opted for the "caller" approach initially and I changed... I probably found a problem with that but I can't remember